core/sequencer/sequencer.go

```diff
@@ -122,15 +122,23 @@ func (s *Sequencer) Initialize(ctx context.Context, logID, mapID int64) error {
 }
 
 // StartSequencingAll starts sequencing processes for all domains.
-func (s *Sequencer) StartSequencingAll(ctx context.Context) error {
-	domains, err := s.admin.List(ctx, false)
-	if err != nil {
-		return fmt.Errorf("admin.List(): %v", err)
-	}
-	for _, d := range domains {
-		go s.StartSigning(ctx, d.LogID, d.MapID, d.MinInterval, d.MaxInterval)
+func (s *Sequencer) StartSequencingAll(ctx context.Context, refresh time.Duration) error {
+	started := make(map[string]bool)
+	ticker := time.NewTicker(refresh)
+
+	for range ticker.C {
+		domains, err := s.admin.List(ctx, false)
+		if err != nil {
+			return fmt.Errorf("admin.List(): %v", err)
+		}
+		for _, d := range domains {
+			if _, ok := started[d.Domain]; !ok {
```

Comment:
this can be simpler (and is less fragile) due to the nil value for bool being 'false':

if !started[d.Domain] {
}

see https://play.golang.org/p/WtCZc7JnGg

Reply:
Awesome 👍